src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/DirectiveCompletionItemProvider.cs

```diff
@@ -116,14 +135,33 @@ internal static List<RazorCompletionItem> GetDirectiveCompletionItems(RazorSynta
             {
                 var completionDisplayText = directive.DisplayName ?? directive.Directive;
                 var commitCharacters = GetDirectiveCommitCharacters(directive.Kind);
+                
                 var completionItem = new RazorCompletionItem(
                     completionDisplayText,
                     directive.Directive,
                     RazorCompletionItemKind.Directive,
-                    commitCharacters: commitCharacters);
+                    commitCharacters: commitCharacters,
+                    isSnippet: false);
                 var completionDescription = new DirectiveCompletionDescription(directive.Description);
                 completionItem.SetDirectiveCompletionDescription(completionDescription);
                 completionItems.Add(completionItem);
+
+                if (s_singleLineDirectiveSnippets.TryGetValue(directive.Directive, out var snippetTexts))
+                {
+                    var snippetCompletionItem = new RazorCompletionItem(
+                        $"{completionDisplayText} ...",
+                        snippetTexts.InsertText,
+                        RazorCompletionItemKind.DirectiveSnippet,
+                        commitCharacters: commitCharacters,
+                        isSnippet: true);
+
+                    var snippetDescription = "@" + snippetTexts.DisplayText
+                                                 + Environment.NewLine + Environment.NewLine
+                                                 + "Type to replace placeholders. [Tab] to navigate or [Return] to complete snippet.";
```

Comment:
This line should probably be localized via resource system. I am not familiar with how it works, so help here would be highly appreciated. I am ok, if you just push commits to my branch

Reply:
You should be able to just the string to the Resources.resx file in this project, build the project, and then it should be available as `Resources.<whatever you named it>`.

Personally I would prefer this line be something like "Inserts a code snippet.\r\nNote: Tab to navigate between elements, Enter to complete." as that matches the style of the existing C# text:
<img width="205" alt="image" src="https://user-images.githubusercontent.com/754264/183815553-5bb343b4-bf9b-47b5-ba1f-a79f05e4ab51.png">

Reply:
@davidwengier  for purity sake you all may eventually want to move this content into the completion resolver side of the world to avoid the serialization costs. Given how few directives there currently are I don't know how much the impact will be but I'd imagine miniscule.

@DoctorKrolic no action for you here, just giving my two cents for the future 😄

Reply:
For me VS doesn't recognize new resource even after solution rebuild. Will see how CI reacts to it though

Reply:
`error CS0117: (NETCORE_ENGINEERING_TELEMETRY=Build) 'Resources' does not contain a definition for 'DirectiveSnippetDescription'` - definitely need some more help here

Reply:
@DoctorKrolic I pushed a one-line change to your PR. Since there are multiple `Resource` classes, it seems that the compiler thought you were referencing a different one. 🙂

Reply:
Thanks!